src/borg/archiver.py

```diff
@@ -108,14 +108,7 @@ def wrapper(self, args, **kwargs):
             with repository:
                 if manifest or cache:
                     kwargs['manifest'], kwargs['key'] = Manifest.load(repository)
-                    # do_recreate uses args.compression is None as in band signalling for "don't recompress",
-                    # note that it does not look at key.compressor. In this case the default compressor applies
-                    # to new chunks.
-                    #
-                    # We can't use a check like `'compression' in args` (an argparse.Namespace speciality),
-                    # since the compression attribute is set. So we need to see whether it's set to something
-                    # true-ish, like a CompressionSpec instance.
-                    if getattr(args, 'compression', False):
+                    if 'compression' in args:
```

Comment:
hmm, isn't compression always there? it has a default (lz4) or it is given explicitly.